### ISSUE TITLE
Replace javax.json 

### ISSUE THREAD

**Issue by dbwiddis** [dblock/oshi#206]
As long as I'm mucking around with JSON, might as well move to a more capable platform.

The original choice to go with JSONP (`javax.json`) was sticking to the java standard to avoid littering the core project with unneeded dependencies.   But with a completely separate json module, I plan to switch to a better library.

Minimum requirements:
 - Null aware (main reason to dump javax.json)
 - Multiple output formats (compact JSON, pretty JSON, and binary)
 - High speed, low memory
 - Supports writing on-the-fly rather than having to create and serialize POJOs.

Considering JSON.simple, GSON, Jackson.  Thoughts:
 - Jackson is most widely used, making it likely that anyone importing oshi json is already using it.  Looks easy to transition code to use its format.  Strongly leaning this way.  
 - GSON is slower for more data but may be the fastest for the small amount of data we have 
 - JSON.simple competes with Jackson for speed on all file sizes.

**Comment by henryx** [comment]
Consider this: objects like `OSFileStore` basically are POJOs, so, extends these objects for use with JSON library is quite simple. In this context, you can view [Flexjson](http://flexjson.sourceforge.net/) too

**Comment by dbwiddis** [comment]
Yeah, we have a few POJOs littered around but then we'd be doing a lot of mix and match.  And considering #202 where I want to use a configuration object to limit the output (e.g., suppress processes or only show top 5) then that just becomes laborious.  I want POJF's (fields) instead...

Which is why a quick peek at Flexjson looks very promising, with their `include()` syntax.  No more manually building arrays of CPU ticks, just include the tick array object.

**Comment by dbwiddis** [comment]
I've spent a good few hours experimenting with the different APIs here.   Unfortunately, most of them make incremental adding of values via methods (the main approach of OSHI rather than POJOs) a little bit more laborious.  There are ways to read values into objects and then serialize the objects, of course (the most obvious just constructing a Map<String,Object>) but this essentially just replicates exactly what the current code is doing, with a different flavored API as the front end.

Basically, these object-serialization focused libraries just don't play nice with OSHI's interface-based `get*()` type of design.  While they'd work well for some of our lower level classes that are POJO, they break down when we get to the higher level interfaces. I could completely rewrite the `oshi-json` version of the "interfaces" as classes to make the serialization easier, but we'd still need code to ensure the values got updated before being serialized.

Basically I don't see a compelling reason to use any implementation that doesn't use the `javax.json` API.  It's part of the Java specification so it has some semi-official API.

The `jackson-javax-json` project looks interesting as an alternate (and better performing) reference implementation to the glassfish one we currently use.  Unfortunately it's currently version 0.9.0 so I'll keep an eye on it for potentially switching when they get to 1.x.  Although Oracle seems to be updating their JSON-P implementation to JSON-P 1.1 as well.

So after a lot of going in circles, I think I'm coming back to our current implementation as sufficient for our needs for now.  It's essentially the same Map view we'd have to implement with any other libraries. Our output is small enough that we probably don't have to be overly concerned about performance.  Most importantly, it's already coded, tested, and stable.  

So I'm tabling this idea for 3.0, reserving the right to revisit it later.
